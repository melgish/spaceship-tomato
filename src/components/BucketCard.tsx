import PHRange from '@/components/PHRange'
import PlantName from '@/components/PlantName'
import type { Bucket } from '@/actions/findBuckets'
import RemoveBucketButton from './RemoveBucketButton'

export default function BucketCard(row: Readonly<Bucket>) {
  return (
    <div className="kv-card">
      <div>Name</div>
      <div>
        {row.name} <RemoveBucketButton row={row} />
      </div>
      <div>Plant</div>
      <div>
        <PlantName plant={row.plant} />
      </div>
      {row.plant && <div>PH</div>}
      {row.plant && (
        <div>
          <PHRange range={row.plant} />
        </div>
      )}
    </div>
  )
}
