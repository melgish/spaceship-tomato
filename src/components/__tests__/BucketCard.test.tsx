import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import BucketCard from '../BucketCard'

// A typical Plant.
const data = {
  id: 1,
  name: 'Test Bucket',
  plantId: 1,
  plant: {
    id: 1,
    name: 'Test Plant',
    maxPH: 7.2,
    minPH: 6.1,
  },
}

describe('when bucket holds a plant', () => {
  it('should render with plant info', () => {
    const { container } = render(<BucketCard {...data} />)

    expect(container).toMatchSnapshot()
  })
})

describe('when bucket does not hold a plant', () => {
  it('should render without plant info', () => {
    const props = { ...data, plantId: 0, plant: null }
    const { container } = render(<BucketCard {...props} />)

    expect(container).toMatchSnapshot()
  })
})
