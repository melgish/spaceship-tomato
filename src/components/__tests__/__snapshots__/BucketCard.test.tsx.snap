// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when bucket does not hold a plant > should render without plant info 1`] = `
<div>
  <div
    class="kv-card"
  >
    <div>
      Name
    </div>
    <div>
      Test Bucket
       
      <button
        class="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
        type="button"
      >
        Remove
      </button>
    </div>
    <div>
      Plant
    </div>
    <div>
      [none]
    </div>
  </div>
</div>
`;

exports[`when bucket holds a plant > should render with plant info 1`] = `
<div>
  <div
    class="kv-card"
  >
    <div>
      Name
    </div>
    <div>
      Test Bucket
       
      <button
        class="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
        type="button"
      >
        Remove
      </button>
    </div>
    <div>
      Plant
    </div>
    <div>
      Test Plant
    </div>
    <div>
      PH
    </div>
    <div>
      6.1
       â€“ 
      7.2
    </div>
  </div>
</div>
`;
